src/Symfony/Component/Yaml/Dumper.php

```diff
@@ -84,6 +84,30 @@ public function dump($input, $inline = 0, $indent = 0, $flags = 0)
             $isAHash = array_keys($input) !== range(0, count($input) - 1);
 
             foreach ($input as $key => $value) {
+                if ($inline - 1 > 0 && Yaml::DUMP_MULTI_LINE_AS_BLOCK & $flags && is_string($value) && false !== strpos($value, "\n")) {
+                    $output .= sprintf(
+                        "%s%s%s |\n",
+                        $prefix,
+                        $isAHash ? Inline::dump($key, $flags).':' : '-',
+                        ''
+                    );
+
+                    foreach (preg_split("/\n|\r\n/", $value) as $row) {
+                        if ('' === $dumpedRow = Inline::doDump($row, true)) {
```

Comment:
actually, I'm wrong, but this is because of a bug in your patch. If a line contains only digits or a numeric pattern, your patch would dump it surrounded by single quotes here, which is wrong